Jupyter Lab 2.1.5 fails to build after installing higlass-jupyter

The system and dependencies:
Mac OS 10.14.6
Python3, node.js and MacFuse installed via Homebrew.
higlass-python (0.4.0) and jupyterlab installed via pip3.

The JupyterLab setup:
JupyterLab v2.1.5
Known labextensions:
   app dir: /usr/local/share/jupyter/lab
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        jupyter-matplotlib v0.7.2  enabled  OK

After running `jupyter labextension install higlass-jupyter`
I get the following error:
```
Building jupyterlab assets (build:prod:minimize)
An error occured.
RuntimeError: JupyterLab failed to build
```
Here is the [error log](https://github.com/higlass/higlass-python/files/4825095/jupyterlab-debug-ddcuq4gz.log). Any help would be greatly appreciated! Cheers.

Thanks for reporting. This might be related to `@jupyter-widgets/base` which was updated to `v3` recently.

The relevant build error seems to be `ModuleNotFoundError: Module not found: Error: Can't resolve 'pixi.js' in '/usr/local/share/jupyter/lab/staging/node_modules/higlass/dist'`

The easiest fix, for now, is to downgrade Jupiter lab until we find a fix.

Okay this is an issue with [higlass](https://github.com/higlass/higlass) itself. Since the hglib version in `package.json` is specified as `^1.8.1` the newest version of hglib will be installed (i.e., `1.9.4` at the time of this writing). I traced the error and found out that in `v1.9.2` an error was introduced which causes the following error during the build (`npm run build` process):

```
Module not found: Error: Can't resolve 'pixi.js' in '/Users/Fritz/Code/4DN/higlass-python/js/node_modules/higlass/dist'
```

@pkerpedjiev Any idea what changed in that version of hglib?

A temporary fix is to fix the version to `v1.9.1` but I wonder if we should rather fix hglib.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,9 @@
+## [v0.4.1](https://github.com/higlass/higlass-python/compare/v0.4.1...v0.4.0)
+
+- Add missing `pixi.js` dependency from `higlass-jupyter`
+- Update `@jupyter-widgets/base` to support Jupyter Lab v2
+- Update simple-httpfs to fix the missing `boto3` error
+
 ## [v0.4.0](https://github.com/higlass/higlass-python/compare/v0.4.0...v0.3.0)
 
 -   API for adding genome position search box within python

--- a/environment.yml
+++ b/environment.yml
@@ -1,0 +1,13 @@
+name: higlass-python
+
+channels:
+  - conda-forge
+  - defaults
+
+dependencies:
+  - python>=3.7
+  - nodejs>=12.*
+  - pip
+  - pip:
+    - jupyterlab>=1.1.1
+    - ipywidgets>=7.5.1

--- a/js/package.json
+++ b/js/package.json
@@ -30,11 +30,12 @@
     "rimraf": "^2.6.1"
   },
   "dependencies": {
-    "@jupyter-widgets/base": "^1 || ^2",
+    "@jupyter-widgets/base": "^1 || ^2 || ^3",
     "css-loader": "^0.28.7",
-    "higlass": "^1.8.1",
+    "higlass": "^1.9.4",
     "higlass-multivec": "^0.2.1",
     "lodash": "^4.17.4",
+    "pixi.js": "^5.3.0",
     "react": "^16.5.1",
     "react-dom": "^16.5.1",
     "style-loader": "^0.19.0"

--- a/requirements.txt
+++ b/requirements.txt
@@ -8,5 +8,5 @@ matplotlib
 multiprocess
 requests
 sh
-simple-httpfs
+simple-httpfs>=0.4.2
 slugid>=2.0.0

